electroncash/interface.py

```diff
@@ -150,15 +135,11 @@ def _get_socket_and_verify_ca_cert(self, *, suppress_errors) -> Tuple[Optional[s
         if s is not None:
             try:
                 context = self.get_ssl_context(cert_reqs=ssl.CERT_REQUIRED, ca_certs=ca_path)
-                s = context.wrap_socket(s, do_handshake_on_connect=True)
-                # validate cert
-                if s and self.check_host_name(s.getpeercert(), self.host):
-                    self.print_error("SSL certificate signed by CA")
-                    # it's good, return the wrapped socket
-                    return s, False
-                # bad cert or other shenanigans, return None but inform caller
-                # to try alternate "pinned self-signed cert" code path
-                return None, False
+                context.check_hostname = True
```

Comment:
This could be an optional argument for  `self.get_ssl_context(..., check_hostname=True)`, just to avoid setting the context attribute twice.

Reply:
Added